loopy/kernel/creation.py

```diff
@@ -1122,7 +1122,10 @@ def __init__(self, domains, instructions, temporary_variables,
         self.default_offset = default_offset
 
         from loopy.symbolic import SubstitutionRuleExpander
-        self.submap = SubstitutionRuleExpander(subst_rules)
+        if subst_rules:
+            self.submap = SubstitutionRuleExpander(subst_rules)
+        else:
+            self.submap = lambda expr: expr
```

Comment:
Instead of repeating this, can we just teach `SubstitutionRuleExpander` to be smarter about no-ops?

Reply:
Done in d6002587d15824c694facdaaaa05759d773a9f93